src/shared/Core/Trace.cs

```diff
@@ -215,13 +216,25 @@ public void Flush()
             {
                 bool isSecretEntry = !(secretKeys is null) &&
                                      secretKeys.Contains(entry.Key, keyComparer ?? EqualityComparer<TKey>.Default);
-                if (isSecretEntry && !this.IsSecretTracingEnabled)
+
+                void WriteSecretLine(object value)
```

Comment:
I wasn't aware you could do this. Interesting!

Reply:
Do you mean the use of [local functions](https://learn.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/local-functions)?

Reply:
Yep!